babe

```diff
@@ -228,6 +233,18 @@ def getAppInfo(appName):
         return None
     return r
 
+def isNodeApp(appName):
+    r = getAppInfo(appName)
```

Comment:
This call is expensive. Can be called if an instance is getting `kill`-ed, but if there are multiple instances of the same thing running on a node (like say nodejs apps), during `decommission`, this will be called multiple times. Optimize if possible by storing it in a `dict`. Call if data is not available in the `dict`.

Reply:
Stored the isNode boolean to a dict and added logic to fetch only if its not already present in the dict

Reply:
The approach is not correct. Create an app info cache. Not a is node app cache. That way, other info can be reused form there as well. Deciding whether it's a node app or not can be done every time and is cheap as it's in memory. All we are trying to do is avoid the network call.